provision/infinispan/create_ispn_clusters.sh

```diff
@@ -0,0 +1,357 @@
+  local cache_name="${3}"
+  local remote_site="${4}"
+  local xsite_mode="${5}"
+  local cr_name=$(echo "${cache_name}" | awk '{print tolower($0)}')
+
+  KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v2alpha1
+kind: Cache
+metadata:
+  name: ${cr_name}
+  namespace: ${namespace}
+spec:
+  clusterName: infinispan
+  name: ${cache_name}
+  template: |-
+    distributedCache:
+      mode: "SYNC"
+      owners: "2"
+      statistics: "true"
+      stateTransfer:
+        chunkSize: 16
+      backups:
+        ${remote_site}:
+          backup:
+            strategy: ${xsite_mode}
+            stateTransfer:
+              chunkSize: 16
+EOF
+}
+
+function deploy_cache_cr_without_cross_site() {
+  local kubecfg="${1}"
+  local namespace="${2}"
+  local cache_name="${3}"
+  local cr_name=$(echo "${cache_name}" | awk '{print tolower($0)}')
+
+  KUBECONFIG="${kubecfg}" oc apply -f - << EOF
+apiVersion: infinispan.org/v2alpha1
+kind: Cache
+metadata:
+  name: ${cr_name}
+  namespace: ${namespace}
+spec:
+  clusterName: infinispan
+  name: ${cache_name}
+  template: |-
+    distributedCache:
+      mode: "SYNC"
+      owners: "2"
+      statistics: "true"
+      stateTransfer:
+        chunkSize: 16
+EOF
+}
+
+function deploy_all_caches() {
+  local kubecfg="${1}"
+  local namespace="${2}"
+  local remote_site="${3}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "sessions" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "actionTokens" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "authenticationSessions" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "offlineSessions" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "clientSessions" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "offlineClientSessions" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "loginFailures" "${remote_site}" "${XSITE_MODE}"
+  deploy_cache_cr "${kubecfg}" "${namespace}" "work" "${remote_site}" "${XSITE_MODE}"
+}
+
+function get_api_url() {
+    local kubecfg="${1}"
+    KUBECONFIG="${kubecfg}" oc config view -o jsonpath='{.clusters[0].cluster.server}' | sed 's|^http[s]://||g'
+}
+
+function create_cross_site_single_cluster() {
+  # Use namespace as site's name
+  local site1="${NS_1}"
+  local site2="${NS_2}"
+  # TLS not required for single cluster cross-site
+  # Login in cluster
+  rosa_oc_login "$KUBECONFIG_1" "${CLUSTER_1}"
+
+  # Check and create the namepsaces if missing
+  KUBECONFIG="${KUBECONFIG_1}" oc new-project "${NS_1}" || true
+  KUBECONFIG="${KUBECONFIG_1}" oc new-project "${NS_2}" || true
+
+  # Deploy an Infinispan cluster in each of the namespaces.
+  deploy_infinispan_cr_same_cluster "${KUBECONFIG_1}" "${NS_1}" "${site1}" "${NS_2}" "${site2}"
+  deploy_infinispan_cr_same_cluster "${KUBECONFIG_1}" "${NS_2}" "${site2}" "${NS_1}" "${site2}"
```

Comment:
Shouldn't here the last ${site2} be ${site1} as defining remoteSite for 2nd cluster, e.g.
`deploy_infinispan_cr_same_cluster "${KUBECONFIG_1}" "${NS_2}" "${site2}" "${NS_1}" "${site1}"`